src/OpenTelemetry.Abstractions/Trace/ITracer.cs

```diff
@@ -50,101 +51,45 @@ public interface ITracer
         // TODO: add sampling hints
 
         /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
-
-        /// <summary>
-        /// Creates root span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartRootSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
-
-        /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
-
-        /// <summary>
-        /// Creates span. If there is active current span, it becomes a parent for returned span.
+        /// Starts root span.
         /// </summary>
         /// <param name="operationName">Span name.</param>
         /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="links">Links collection.</param>
-        /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, SpanKind kind, DateTimeOffset startTimestamp, IEnumerable<Link> links);
-
-        /// <summary>
-        /// Creates span.
-        /// </summary>
-        /// <param name="operationName">Span name.</param>
-        /// <param name="parent">Parent for new span.</param>
-        /// <param name="kind">Kind.</param>
-        /// <param name="startTimestamp">Start timestamp.</param>
-        /// <param name="linksGetter">Func to retrieve the Links collection.</param>
+        /// <param name="options">Advanced span creation options.</param>
         /// <returns>Span instance.</returns>
-        ISpan StartSpan(string operationName, ISpan parent, SpanKind kind, DateTimeOffset startTimestamp, Func<IEnumerable<Link>> linksGetter);
+        ISpan StartRootSpan(string operationName, SpanKind kind, SpanCreationOptions options);
```

Comment:
Could these two methods become extensions also?

Reply:
StartSpan can and will in a different PR. StartRootspan has special meaning - ignore any implicit parents which could be expressed as a flag or in the method name, so I don't see a good option on how to move it to extensions. Let me know if you have any solution in mind